fluent-bundle/src/bundle.rs

```diff
@@ -243,7 +251,44 @@ impl<R> FluentBundle<R> {
     where
         R: Borrow<FluentResource>,
     {
-        self.get_message(id).is_some()
+        self.get_entry_message(id).is_some()
+    }
+
+    pub fn get_message(&self, id: &str) -> Option<Message>
+    where
+        R: Borrow<FluentResource>,
+    {
+        let message = self.get_entry_message(id)?;
+        let value = message.value.as_ref();
+        let mut attributes = if message.attributes.is_empty() {
+            HashMap::new()
+        } else {
+            HashMap::with_capacity(message.attributes.len())
+        };
+
+        for attr in message.attributes.iter() {
+            attributes.insert(attr.id.name, &attr.value);
+        }
+        return Some(Message { value, attributes });
```

Comment:
Would it make more sense to do all this prep work in `add_resource` instead?

Reply:
I'd prefer not to, at least for now.